File: ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal: true

# Squares
class Squares
  def initialize(number)
    @num = number
  end

  def square_of_sum
    (1..@num).sum.pow(2)
  end

  def sum_of_squares
    (1..@num).map { |e| e.pow(2) }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: erlang/difference-of-squares/src/difference_of_squares.erl
-module(difference_of_squares).

-export([difference_of_squares/1, square_of_sum/1, sum_of_squares/1]).


difference_of_squares(Number) ->
    square_of_sum(Number) - sum_of_squares(Number).

sum(0) -> 0;

sum(N) ->
    N + sum(N-1).

square_of_sum(Number) ->
    S = sum(Number),
    S*S.

sum_of_squares(0) -> 0;

sum_of_squares(Number) ->
    Number * Number + sum_of_squares(Number - 1 ).
